src/CMakeLists.txt

```diff
@@ -538,87 +540,60 @@ set(zeek_SRCS
 collect_headers(zeek_HEADERS ${zeek_SRCS})
 
 add_library(zeek_objs OBJECT ${zeek_SRCS})
+target_link_libraries(zeek_objs PRIVATE $<BUILD_INTERFACE:zeek_internal>)
+add_dependencies(zeek_objs zeek_autogen_files)
+add_clang_tidy_files(${zeek_SRCS})
+zeek_target_link_libraries(zeek_objs)
 
-if (ZEEK_STANDALONE)
-    add_executable(zeek main.cc
-                $<TARGET_OBJECTS:zeek_objs>
-                ${zeek_HEADERS}
-                ${bro_SUBDIR_LIBS}
-                ${bro_PLUGIN_LIBS}
-    )
+if (TARGET zeek_exe)
+    target_sources(zeek_exe PRIVATE main.cc ${zeek_HEADERS})
 
     # npcap/winpcap need to be loaded in delayed mode so that we can set the load path
     # correctly at runtime. See https://npcap.com/guide/npcap-devguide.html#npcap-feature-native
     # for why this is necessary.
     if ( MSVC AND HAVE_WPCAP )
         set(zeekdeps ${zeekdeps} delayimp.lib)
-        set_target_properties(zeek PROPERTIES LINK_FLAGS "/DELAYLOAD:wpcap.dll")
+        set_target_properties(zeek_exe PROPERTIES LINK_FLAGS "/DELAYLOAD:wpcap.dll")
     endif()
 
-    target_link_libraries(zeek ${bro_PLUGIN_LINK_LIBS} ${zeekdeps} ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS})
+    target_link_libraries(zeek_exe PRIVATE ${zeekdeps} ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS})
 
     # Export symbols from zeek executable for use by plugins
-    set_target_properties(zeek PROPERTIES ENABLE_EXPORTS TRUE)
+    set_target_properties(zeek_exe PROPERTIES ENABLE_EXPORTS TRUE)
 
     if ( MSVC )
         set(WINDOWS_EXPORT_ALL_SYMBOLS ON)
     endif ()
 
-    install(TARGETS zeek RUNTIME DESTINATION bin)
-
-    set(BRO_EXE zeek
-        CACHE STRING "Zeek executable binary" FORCE)
-
-    set(BRO_EXE_PATH ${CMAKE_CURRENT_BINARY_DIR}/zeek
-        CACHE STRING "Path to Zeek executable binary" FORCE)
+    target_include_directories(zeek_exe PRIVATE
```

Comment:
This whole section feels sort of obnoxious. We really have header files in all of these directories?

Reply:
Good question. So far I've been just porting things over without trying to minimize them. Let me try if some are redundant.